### ISSUE TITLE
lua sockets library loaded in non-standard fashion

### ISSUE THREAD

**Issue by patrickmeehan** [moai/moai-dev#157]
http://getmoai.com/forums/post2551.html#p2551

**Comment by patrickmeehan** [comment]
Odie compiled the Lua code that comes with Lua socket and packed it into C files. He also changed the way the loading works to play nice with Lua's module system. I did not have time to research this, but I did bring it over to Visual Studio and Android.

Odie's version is in 3rd party/luasocketEmbed. I was lazy, so I copied it whole cloth into 3rd party/luasocket-2.0.2/src. Doing it that way meant I didn't have to add any include paths. The problem is that if we update loasocket we'll overwrite the embed. So a better solution would be to rename Odie's external folder to something like luasocket-2.0.2-embed to make the versioning clear and also show that the files he produced are not part of the luasocket distro.

Since Xcode uses Odie's and the other platforms use mine and neither setup is ideal we need a new, consistent place for the 'embed' to live and to update all the projects to support it. We also need to add a readme and maybe some scripts to re-generate the embed. Would be nice to have a documented process for embedding 3rd party Lua so we can more easily add extension libraries.
